### ISSUE TITLE
Lint/UnusedMethodArgument should say "If it's unnecessary, remove it."

### ISSUE THREAD

**Issue by JunichiIto** [rubocop/rubocop#11322]
## Is your feature request related to a problem? Please describe.

I have a code like this:

```ruby
def double(number, other)
  number * 2
end
```

Then, RuboCop warns:

```
foo.rb:17:20: W: [Correctable] Lint/UnusedMethodArgument: Unused method argument - other. If it's necessary, use _ or _other as an argument name to indicate that it won't be used.
def double(number, other)
                   ^^^^^
```

If I run `rubocop -A`, it will be fixed like this:

```ruby
def double(number, _other)
  number * 2
end
```

However, that is not a good fix. The second argument should be removed in this case:

```ruby
def double(number)
  number * 2
end
```

## Describe the solution you'd like

RuboCop should determine the argument is necessary or not. If it's necessary, it would be okay to rename to `_other`. But if not, it should be removed instead of renaming it.

However, I guess it would be tough to say necessary or not for RuboCop. At least the message should be changed like this:

> If it's necessary, use _ or _other as an argument name to indicate that it won't be used. If it's unnecessary, remove it.

## Additional context

As a programming mentor, I've seen unnecessary arguments like `_others` so many times since newbies tend to apply `rubocop -A` blindly. It might be good idea to disable auto-correct by default because they should consider by themselves to rename or remove warned arguments.

**Comment by koic** [comment]
It shouldn't break existing autocorrection behavior for users who expect it. It can disable it by setting the following on the user side .rubocop.yml:

```yaml
Lint/UnusedMethodArgument:
  AutoCorrect: false
```

**Comment by JunichiIto** [comment]
> It shouldn't break existing default autocorrection behavior for users who expect it.

I understand your thoughts. But I'm still doubtful about the current behavior. I think RuboCop can edit code in wrong way via auto-correction.

By the way, I found this line:

> Currently there might still be cops that aren’t marked as unsafe or with unsafe autocorrect.
>
> https://docs.rubocop.org/rubocop/usage/autocorrect.html

Shouldn't we say Lint/UnusedMethodArgument is unsafe too?

**Comment by ydah** [comment]
> - Safe (true/false) - indicates whether the cop can yield false positives (by design) or not.
> - SafeAutoCorrect (true/false) - indicates whether the autocorrect a cop does is safe (equivalent) by design. If a cop is unsafe its autocorrect automatically becomes unsafe as well.

https://docs.rubocop.org/rubocop/usage/autocorrect.html#safe-autocorrect

I think it is different to say that this cop is Unsafe.
We say the cop is Unsafe only when there are cases of automatic modifications or false positives that change the semantics of the code.

So I think what we can do is add documents and change messages.
Or you may join the discussion in the following Issue.

- https://github.com/rubocop/rubocop/issues/11251

**Comment by koic** [comment]
Yup, it is probably for safety that method signature is not changed. So this cop can safely detect and autocorrection.
Whether or not to remove unused arguments is a matter of user application design and can be supplemented with the documentation.

**Comment by JunichiIto** [comment]
Thank you for your information. I noticed there were similar discussions:

- https://github.com/rubocop/rubocop/issues/11251
- https://github.com/rubocop/rubocop/issues/11199

Aside from agreeing or not, this line is interesting for me:

> Historically none of them did any autocorrection, as usually there were subtleties that required human input, but we've gradually started to add more and more on that front due to a lot of demand from our users - many people said they were fine with the implications as long as this reduced the manual work for them to some extent.

And [my PR](https://github.com/rubocop/rubocop/pull/11323) might help this:

> So I think what we can do is add documents and change messages.
